pkg/server/router/credential_test.go

```diff
@@ -601,4 +601,244 @@ func TestCredentialRouter(t *testing.T) {
+		assert.NotEmpty(tt, credentialSubjectAfterRevoke)
+
+		encodedListAfterSuspended := credentialSubjectAfterRevoke["encodedList"]
+		assert.NotEmpty(tt, encodedListAfterSuspended)
+
+		assert.NotEqualValues(tt, encodedListAfterSuspended, encodedList)
+	})
+
+	t.Run("Create Suspendable and Revocable Credential Should Be Error", func(tt *testing.T) {
+		issuer, schema, credService := createCredServicePrereqs(tt)
+		subject := "did:test:345"
+
+		createdCred, err := credService.CreateCredential(context.Background(), credential.CreateCredentialRequest{
+			Issuer:     issuer,
+			Subject:    subject,
+			JSONSchema: schema,
+			Data: map[string]any{
+				"email": "Satoshi@Nakamoto.btc",
+			},
+			Expiry:      time.Now().Add(24 * time.Hour).Format(time.RFC3339),
+			Revocable:   true,
+			Suspendable: true,
+		})
+
+		assert.Error(tt, err)
+		assert.ErrorContains(tt, err, "credential cannot be revocable and suspendable")
+		assert.Empty(tt, createdCred)
+	})
+
+	t.Run("Update Suspendable and Revocable Credential Should Be Error", func(tt *testing.T) {
+		issuer, schema, credService := createCredServicePrereqs(tt)
+		subject := "did:test:345"
+
+		createdCred, err := credService.CreateCredential(context.Background(), credential.CreateCredentialRequest{
+			Issuer:     issuer,
+			Subject:    subject,
+			JSONSchema: schema,
+			Data: map[string]any{
+				"email": "Satoshi@Nakamoto.btc",
+			},
+			Expiry:      time.Now().Add(24 * time.Hour).Format(time.RFC3339),
+			Suspendable: true,
+		})
+
+		assert.NoError(tt, err)
+		assert.NotEmpty(tt, createdCred)
+
+		updatedStatus, err := credService.UpdateCredentialStatus(context.Background(), credential.UpdateCredentialStatusRequest{ID: createdCred.ID, Revoked: true, Suspended: true})
+		assert.Nil(tt, updatedStatus)
+		assert.Error(tt, err)
+		assert.ErrorContains(tt, err, "cannot update both suspended and revoked status")
+	})
+
+	t.Run("Update Suspended On Revoked Credential Should Be Error", func(tt *testing.T) {
+		issuer, schema, credService := createCredServicePrereqs(tt)
+		subject := "did:test:345"
+
+		createdCred, err := credService.CreateCredential(context.Background(), credential.CreateCredentialRequest{
+			Issuer:     issuer,
+			Subject:    subject,
+			JSONSchema: schema,
+			Data: map[string]any{
+				"email": "Satoshi@Nakamoto.btc",
+			},
+			Expiry:    time.Now().Add(24 * time.Hour).Format(time.RFC3339),
+			Revocable: true,
+		})
+
+		assert.NoError(tt, err)
+		assert.NotEmpty(tt, createdCred)
+
+		updatedStatus, err := credService.UpdateCredentialStatus(context.Background(), credential.UpdateCredentialStatusRequest{ID: createdCred.ID, Suspended: true})
+		assert.Nil(tt, updatedStatus)
+		assert.Error(tt, err)
+		assert.ErrorContains(tt, err, "has a different status purpose<revocation> value than the status credential<suspension>")
+	})
+
+}
+
+func createCredServicePrereqs(tt *testing.T) (string, string, credential.Service) {
+	bolt := setupTestDB(tt)
+	assert.NotNil(tt, bolt)
```

Comment:
nit: you may want to change these asserts to `require` in setup so that the test fails fast if the prereqs aren't there